feat(linter): support class sorting for `tw.div`

Yz4230 opened the pull request:
<!--  
    Thanks for submitting a Pull Request! We appreciate you taking the time to contribute to this project.  
    Please provide enough information so that others can review your PR.  
    Once created, your PR will be automatically labeled according to the changed files.  
    Learn more about contributing: https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md  
-->

## Summary

<!--  
    Explain the **motivation** for making this change.  
    What existing problem does the pull request solve?  
-->

This PR adds support for class sorting in the `tw.div` syntax, commonly used with libraries such as [twin.macro](https://github.com/ben-rogerson/twin.macro) and [react-twc](https://github.com/gregberge/twc). 

```diff
- tw.div`px-2·foo·p-4·bar`;
+ tw.div`foo·bar·p-4·px-2`;
```

To enable sorting for `tw.div` or similar patterns, need to include a configuration entry like `tw.*` in `options.functions`.

<!--  
    Link any relevant issues if necessary or include a transcript of any Discord discussion.  
-->

## Test Plan

<!--  
    What demonstrates that your implementation is correct?  
-->

- Updated snapshots for `useSortedClasses` tests to reflect the added support for the `tw.div` syntax.

ematipico reviewed (changes requested):
Thank you for your contribution! Three things are missing in this PR: 
- A new changelog line that explains the new **feature**
- Update the documentation, and explain the new feature. You should add something like `**Since v2.0.0**`
- Update the documentation and add a new example with this new feature. The contribution guide [explains how to do so](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#using-rule-options) 

Also, one question: what was the reason for adding `tw.*` not `tw.div`?

chansuke commented on crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/options.rs:
```diff
@@ -33,6 +33,25 @@ impl UtilityClassSortingOptions {
         false
     }
 
+    pub(crate) fn match_function(&self, name: &str) -> bool {
+        let matchers = self.functions.iter().flatten();
+        let parts = name.split('.');
+        for matcher in matchers {
+            let mut matcher = matcher.split('.');
+            let mut parts = parts.clone();
+
+            let mut zip = matcher.by_ref().zip(parts.by_ref());
+            if zip.all(|(m, p)| m == "*" || m == p)
+                && matcher.next().is_none()
+                && parts.next().is_none()
+            {
+                return true;
+            }
+        }
+
+        false
+    }
```
```suggestion
    pub(crate) fn match_function(&self, name: &str) -> bool {
        self.functions.iter().flatten().any(|matcher| {
            let mut matcher_parts = matcher.split('.');
            let mut name_parts = name.split('.').clone();

            let all_parts_match = matcher_parts
                .by_ref()
                .zip(name_parts.by_ref())
                .all(|(m, p)| m == "*" || m == p);

            all_parts_match
                && matcher_parts.next().is_none()
                && name_parts.next().is_none()
        })
    }
```
uses .any() to replace the explicit loop and manual return true. the result is functionally equivalent, with the same time and space complexity.
I prepared some testcases and  tested this on the playground, and the behavior matches the original implementation.

Yz4230 replied:
Thank you for suggesting this!🥹
Committed without using redundant `.clone()`! [2e70bca](https://github.com/biomejs/biome/pull/4699/commits/2e70bca310e62695e8ff4c9bcf20890fa38afd7f)

```diff
-            let mut name_parts = name.split('.').clone();
+            let mut name_parts = name.split('.');
```

Yz4230 commented:
@ematipico 
Thank you very much for your review!
I added the changelog and updated the document.

> Also, one question: what was the reason for adding `tw.*` not `tw.div`?

This is for supporting tags other than `div` such as `tw.span` or `tw.a`.

Yz4230 commented:
@ematipico 

Thank you for your review and approve!✨

Which version is this PR planned to be included in?
I marked it as v2.0.0 in the Files changed, but since there are no breaking changes, I think it could be released earlier.

If there are any shortcomings in this PR, I would appreciate it if you could point them out👀

dyc3 commented:
The next version we release will likely be 2.0. We aren't planning any minor releases before that.

Yz4230 commented:
Nice! Thank you!
